brickBreaker.js

```diff
@@ -0,0 +1,217 @@
+class BrickBreaker {
+  constructor() {
+    this.canvas = document.getElementById("canvas");
+    this.ctx = this.canvas.getContext("2d");
+    this.initializeEntities();
+    this.initializeEventListeners();
+  }
+
+  initializeBricks() {
+    this.bricks = [];
+    this.brickCols = 5;
+    this.brickRows = 8;
+    const brickWidth = 50;
+    const brickLength = 25;
+
+    const colors = ["red", "blue", "green", "yellow", "purple"];
+
+    for (let col = 0; col < this.brickCols; col++) {
+      for (let row = 0; row < this.brickRows; row++) {
+        if (!this.bricks[row]) {
+          this.bricks[row] = [];
+        }
+        const x = brickWidth * row;
+        const y = brickLength * col;
+        const brick = new Brick(
+          this.ctx,
+          x,
+          y,
+          brickWidth,
+          brickLength,
+          colors[col]
+        );
+        this.bricks[row][col] = brick;
+      }
+    }
+  }
+
+  initializeEntities() {
+    this.score = 0;
+    this.ball = new Ball(this.ctx, this.canvas.width / 2, 500, 10, "#FFF", 4);
+    this.paddle = new Paddle(
+      this.canvas,
+      this.ctx,
+      100,
+      10,
+      150,
+      525,
+      5,
+      "grey"
+    );
+    this.initializeBricks();
+  }
+
+  initializeEventListeners() {
+    window.addEventListener("keydown", this.keyDownHandler.bind(this));
+    window.addEventListener("keyup", this.keyUpHandler.bind(this));
+  }
+
+  keyDownHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(-1);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(1);
+        break;
+      case " ":
+        this.reset();
+        break;
+    }
+  }
+
+  keyUpHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(0);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(0);
+        break;
+    }
+  }
+
+  updateEntities() {
+    this.paddle.move();
+    this.ball.move();
+  }
+
+  detectCanvasCollision(ballTop, ballLeft, ballRight) {
+    if (ballLeft <= 0 || ballRight >= this.canvas.width) {
+      this.ball.invertXDirection();
+    } else if (ballTop <= 0) {
+      this.ball.invertYDirection();
+    } else if (ballTop > this.canvas.height) {
+      this.ball.xDirection = 0;
+      this.ball.yDirection = 0;
+    }
+  }
+
+  detectBallCollisionWithEntity(
+    entity,
+    ballTop,
+    ballBottom,
+    ballLeft,
+    ballRight
+  ) {
+    if (entity === this.paddle) {
```

Comment:
whats the point of splitting this function is you're just going to check typing of the entity? you're supposed to pass in ball and entity and do collision detection on them there. you are repeating logic still by simply shifting logic here